test/test-lib/api-helpers.ts

```diff
@@ -81,33 +82,45 @@ export const addImageToRelease = async (
 		.expect(201);
 };
 
-export const expectResourceToMatch = async (
-	pineUser: PineTest,
+export const expectResourceToMatch = async <T = AnyObject>(
+	pineUser: PineTest | PinejsClient | { token: string } | string,
 	resource: string,
 	id: number | AnyObject,
 	expectations: Dictionary<
 		| null
 		| string
 		| number
 		| boolean
-		| ((chaiPropertyAssetion: Chai.Assertion) => void)
+		| object
+		| ((chaiPropertyAssertion: Chai.Assertion) => void)
 	>,
-) => {
-	const { body: result } = await pineUser
-		.get({
-			resource,
-			id,
-			options: {
-				$select: Object.keys(expectations),
-			},
-		})
-		.expect(200);
-	for (const [key, valueOrAssertion] of Object.entries(expectations)) {
-		if (typeof valueOrAssertion === 'function') {
-			valueOrAssertion(expect(result).to.have.property(key));
+): Promise<T> => {
+	if (typeof pineUser === 'string' || 'token' in pineUser) {
+		pineUser = pineTest.clone({ passthrough: { user: pineUser } });
+	}
+
+	const pineClientRequest = pineUser.get({
+		resource,
+		id,
+		options: {
+			$select: Object.keys(expectations),
+		},
+	});
+	const result = (
+		'expect' in pineClientRequest
+			? (await (pineClientRequest as ReturnType<PineTest['get']>).expect(200))
+					.body
+			: await pineClientRequest
+	) as T | undefined;
+	expect(result).to.be.an('object');
+	Object.entries(expectations).forEach(([key, valueOrAssetion]) => {
```

Comment:
Why is that changed from the `for ([] of ) {}` ?

Reply:
copied over from bC-api, will make it a for-of again :+1: